File: 1894-find-the-student-that-will-replace-the-chalk/1894-find-the-student-that-will-replace-the-chalk.cpp
class Solution {
public:
    int chalkReplacer(vector<int>& chalk, int k) {
        long sum = 0;
        for (int num : chalk) {
            sum += num;
        }
        k %= sum;
        for (int i = 0; i < chalk.size(); i++) {
            if (k < chalk[i])
                return i;
            k -= chalk[i];
        }
        return chalk.size() - 1;
    }
};

File: 1894-find-the-student-that-will-replace-the-chalk/1894-find-the-student-that-will-replace-the-chalk.go
func chalkReplacer(chalk []int, k int) int {
	sum := 0
	for _, num := range chalk {
		sum += num
	}
	k %= sum
	for i, num := range chalk {
		if k < num {
			return i
		}
		k -= num
	}
	return len(chalk) - 1
}